From 0047b5d36ae50af7a703c7d9a4cf979dd6e746e5 Mon Sep 17 00:00:00 2001
From: Evan Huus <evan.huus@jadedpixel.com>
Date: Tue, 17 Jun 2014 21:43:49 +0000
Subject: [PATCH 3/4] Set the rlc len field before we try to use it in an
 expert info.

Conflicts:
	epan/dissectors/packet-rlc.c

Bug:9795
Change-Id: I740a9fe36278995cf6e590206f600c037789f0eb
Reviewed-on: https://code.wireshark.org/review/2344
Reviewed-by: Evan Huus <eapache@gmail.com>
(cherry picked from commit ba6eb5c72ffe82ca0e51c7083240975a5b118ad2)
Reviewed-on: https://code.wireshark.org/review/2345
(cherry picked from commit 2c877e0c77be47f5ac78b8f307081b5e4f36b2be)
Reviewed-on: https://code.wireshark.org/review/2346
Reviewed-on: https://code.wireshark.org/review/3439
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
---
 epan/dissectors/packet-rlc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-rlc.c b/epan/dissectors/packet-rlc.c
index 1045b83..0cb5d6f 100644
--- a/epan/dissectors/packet-rlc.c
+++ b/epan/dissectors/packet-rlc.c
@@ -1156,6 +1156,7 @@ rlc_decode_li(enum rlc_mode mode, tvbuff_t *tvb, packet_info *pinfo, proto_tree
 					* may not be larger than the total remaining length and no
 					* LI may be smaller than its preceding one
 					*/
+					li[num_li].len = li[num_li].li - prev_li;
 					if (((li[num_li].li > total_len) && !global_rlc_headers_expected)
 						|| (li[num_li].li < prev_li)) {
 						/* add malformed LI for investigation */
@@ -1167,7 +1168,6 @@ rlc_decode_li(enum rlc_mode mode, tvbuff_t *tvb, packet_info *pinfo, proto_tree
 						col_append_str(pinfo->cinfo, COL_INFO, "[Malformed Packet]");
 						return -1; /* just give up on this */
 					}
-					li[num_li].len = li[num_li].li - prev_li;
 					prev_li = li[num_li].li;
 			}
 		}
-- 
2.0.1

